docs/atomic-swap-protocol-v1.md

```diff
@@ -0,0 +1,155 @@
+  amount of tokens offered, the hash copied from Alice's offer and a timeout.
+  This contract locks his tokens and sends them to a<sub>Y</sub> if the preimage
+  is releaved on-chain.
+- (Variant 1): Alice now uses her preimage to claim Bob's swap offer on Y,
+  causing the token transfer from b<sub>Y</sub> to a<sub>Y</sub>. Bob looks up
+  this now public preimage on Y to claim Alice's swap offer on X, causing the
+  token transfer from a<sub>X</sub> to b<sub>X</sub>.
+- (Variant 2): Alice decides not to reveal her preimage before the timeout of
+  Bob's offer is reached. As soon as the timeout is reached, Bob aborts his
+  offer and gets back his tokens. Alice also needs to wait for the timeout of
+  her offer to be reached in order to abort and get back her tokens.
+
+## Actor positions
+
+As soon as both offers are created, the actor that created the preimage has the
+one-sided option to execute the atomic swap or to let it expire. Thus this actor
+is called to be _in long position_. The other actor is called to be _in short
+position_.
+
+Being in long position has two advantages:
+
+1. having the free option to execute the swap or not and
+2. claiming the other actor's swap offer first, which reduces the risk of
+   loosing tokens in the case of a half executed swap.
+
+Being in short position has that advantage to be able to review the other
+actor's swap offer before creating an offer oneself.
+
+## The swap offer
+
+A swap offer is a HTLC with the following properties:
+
+- The amount of tokens is fixed. Implementation may support a group of multiple
+  amounts.
+- The recipient address is fixed.
+- Creating the contract locks the funds.
+- A hash of the preimage is fixed.
+- A timeout expressed as absolute time or block height is fixed.
+- At any point in time before the timeout, the swap offer can be claimed.
+- As soon as the timeout is reached, the funds can be returned to the creator.
+
+Common synonyms: _swap_, _contract_, _escrow_.
+
+### Preimage and hashing
+
+The preimage is 32 bytes of raw binary data. It is locally created by the long
+position and must be kept a secret. It is in the creator's own interest to use a
+sufficiently good source of entropy to generate it, but this is not required by
+the protocol.
+
+The hashing algorithm is fixed to SHA-256. The hash is 32 bytes of raw binary
+data with `hash = sha256(preimage)`.
+
+### Timeouts
+
+TODO
+
+### Swap offer actions
+
+Actions are mutations of the on-chain state triggered by transactions.
+
+- **Create:** By creating a swap offer, the funds are locked in the contract. A
+  timeout and a hash must be specified.
+- **Claim:** By revealing a preimage for the hash before the timeout is reached,
+  the contract transfers the tokens to the recipient. Requires a preimage of
+  exactly 32 bytes. Common synonyms: _release_, _execute_.
+- **Abort:** Withdraws the tokens from the contract to the creator. Requires
+  timeout to be reached.
+
+### Swap offer open state
```

Comment:
It's confusing that "open" is used for this state category and one of the possible positions.

Reply:
Yes I was also thinking about this issue. Do you have a suggestion for a change? Either for the state name or the state value.

Reply:
Best I could come up with is just calling this the "swap offer state" and calling the next section "swap offer expiration" (not mentioning state). Otherwise, "swap offer process state"?

Reply:
I prefer "swap offer process state" since both are states of the swap offer. Process state will do the job

Reply:
Do you think we should rename the enum in the contract then?

Reply:
Yes. Not super urgent but I think we should do it.

I will also rename this client-side enum: https://github.com/iov-one/iov-core/blob/atomic-swap-doc/packages/iov-bcp/src/atomicswaptypes.ts#L7